lib/extras/enc/pnm.cc

```diff
@@ -69,7 +69,7 @@ class PNMEncoder : public Encoder {
                                     std::vector<uint8_t>* bytes) const = 0;
 };
 
-class PPMEncoder : public PNMEncoder {
+class PPMEncoder : public BasePNMEncoder {
```

Comment:
I think it would be clearer if this class would be PNMEncoder, inherited from BasePNMEncoder (since its EncodeImage already supports both ppm and pgm), with the AcceptedFormats() supporting 1 or 3 channels.

And then we would have two classes derived from this, PPMEncoder, PGMEncoder each overriding only AcceptedFormats()

Reply:
yes, done